### PR TITLE
final-step-of-project

### PR DESCRIPTION
**Pull request by mburko**
Project was comoplited and ready for review

### CODE REVIEW

File: Task/src/test/java/com/softserve/firsttask/data/City.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    public String getLongtitude() {
+        return longtitude;
+    }
+
+    public void setLongtitude(String longtitude) {
+        this.longtitude = longtitude;
+    }
+
+    //
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/data/City.java]: Why we need this?

Comment (mburko) [reply Task/src/test/java/com/softserve/firsttask/data/City.java]: It was fixed already

File: Task/src/test/java/com/softserve/firsttask/data/PlacesRepository.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.List;
+
+import com.softserve.firsttask.tools.ExcelReader;
+
+public final class PlacesRepository {
+
+   private PlacesRepository() {
+   }
+
+   public static City getKyiv() {
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/data/PlacesRepository.java]: We are not using getKyiv, getLviv, getIvanoFrankivsk, so may be deleted.

File: Task/src/test/java/com/softserve/firsttask/pages/HeaderPart.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+    public WebElement getLatestPlaces() {
+        return latestPlaces;
+    }
+
+    public String getLatestPlacesText() {
+        return getLatestPlaces().getText();
+    }
+
+    public void clicklatestPlaces() {
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/pages/HeaderPart.java]:
```suggestion
    public void clickLatestPlaces() {
```

File: Task/src/test/java/com/softserve/firsttask/tests/SmokeTest.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    public void pageObjectFunctionalTest() {
+        HomePage homePage = loadApplication()
+                .findByPlaceName("Kyiv");
+    }
+
+    @Test
+    public void externalReaderTest() {
+        System.out.printf(PlacesRepository.fromExcel("places.xlsx").get(2).getName());
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/tests/SmokeTest.java]: What we are checking here?

File: Task/src/test/java/com/softserve/firsttask/tools/ExternalReader.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+public interface ExternalReader {
+    String FILE_NOT_FOUND_EXCEPTION = "File %s could not be found";
+    String FILE_NOT_READ_EXCEPTION = "File %s could not be read";
+    String FILE_NOT_CLOSE_EXCEPTION = "File %s could not be closed";
+    String DB_READING_ERROR = "DB Reading Error, %s";
+
+    List<List<String>> getAllCells(); // public abstract
+
+    List<List<String>> getAllCells(String path);
+
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/tools/ExternalReader.java]: remove whitespace

File: Task/src/test/java/com/softserve/firsttask/tools/ListUtils.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+   public static Object[][] toMultiArray(List<?> list, Object object) {
+      Object[][] array = new Object[list.size()][2];
+      for (int i = 0; i < list.size(); i++) {
+         array[i][0] = list.get(i);
+         array[i][1] = object;
+      }
+      return array;
+   }
+
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/tools/ListUtils.java]: remove whitespace

File: Task/src/test/java/com/softserve/firsttask/pages/HomePage.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        return getDescription().getText();
+    }
+
+    // grayBox
+
+    public WebElement getGrayBox() {
+        return grayBox;
+    }
+
+    public GrayBoxComponent getGrayBoxComponent() {
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/pages/HomePage.java]: Maybe we can give it a more meaningful name?
smth like: GrayBoxComponent - > SearchPlaceComponent

File: Task/src/test/java/com/softserve/firsttask/pages/HeaderPart.java [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+        return getLatLongToUtm().getText();
+    }
+
+    public void clickLatLongToUtm() {
+        getLatLongToUtm().click();
+    }
+
+    // Functional
+
+    public HomePage gotoHomePage() {
```
Comment (ghost) [inline Task/src/test/java/com/softserve/firsttask/pages/HeaderPart.java]:
```suggestion
    public HomePage goToHomePage() {
```

**Review by ghost — CHANGES_REQUESTED** [review]
The main data provider logic needs to be reworked. All data from .xls need to be pulled somehow automatically... In case we have 100000 values in file...

Also, I'm unable to run the tests. Please update instructions how to run them

`Volodymyrs-MacBook-Pro:Task volodymyrparlah$ mvn test
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: osx
[INFO] os.detected.arch: x86_64
[INFO] os.detected.version: 10.14
[INFO] os.detected.version.major: 10
[INFO] os.detected.version.minor: 14
[INFO] os.detected.classifier: osx-x86_64
[INFO] 
[INFO] -----------------< com.softserve.firsttask:FirstTask >------------------
[INFO] Building FirstTask 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ FirstTask ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/volodymyrparlah/Documents/Learning/ssFirstTask/Task/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ FirstTask ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ FirstTask ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.0:testCompile (default-testCompile) @ FirstTask ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- webdriverextensions-maven-plugin:3.1.3:install-drivers (pre-test) @ FirstTask ---
[INFO] Installation directory "/Users/volodymyrparlah/Documents/Learning/ssFirstTask/Task/target/test-classes"
[INFO] Installing drivers from configuration
[INFO] chromedriver-osx-32bit version 75.0.3770.90
[INFO]   Already installed
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ FirstTask ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running TestSuite
Starting ChromeDriver 75.0.3770.90 (a6dcaf7e3ec6f70a194cc25e8149475c6590e025-refs/branch-heads/3770@{#1003}) on port 22137
Only local connections are allowed.
Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
[ERROR] Tests run: 10, Failures: 1, Errors: 0, Skipped: 9, Time elapsed: 13.548 s <<< FAILURE! - in TestSuite
[ERROR] beforeClass(com.softserve.firsttask.tests.ValuesTest)  Time elapsed: 2.856 s  <<< FAILURE!
org.openqa.selenium.SessionNotCreatedException: 
session not created: This version of ChromeDriver only supports Chrome version 75
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'Volodymyrs-MacBook-Pro.local', ip: 'fe80:0:0:0:142b:cba8:9912:2916%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.14.4', java.version: '1.8.0_181'
Driver info: driver.version: ChromeDriver
remote stacktrace: 0   chromedriver-osx-32bit              0x000000010fe74319 chromedriver-osx-32bit + 3597081
1   chromedriver-osx-32bit              0x000000010fe03f73 chromedriver-osx-32bit + 3137395
2   chromedriver-osx-32bit              0x000000010fbafc7f chromedriver-osx-32bit + 695423
3   chromedriver-osx-32bit              0x000000010fb1da6b chromedriver-osx-32bit + 96875
4   chromedriver-osx-32bit              0x000000010fb19b18 chromedriver-osx-32bit + 80664
5   chromedriver-osx-32bit              0x000000010fb46139 chromedriver-osx-32bit + 262457
6   chromedriver-osx-32bit              0x000000010fb43753 chromedriver-osx-32bit + 251731
7   chromedriver-osx-32bit              0x000000010fb1faea chromedriver-osx-32bit + 105194
8   chromedriver-osx-32bit              0x000000010fb20b45 chromedriver-osx-32bit + 109381
9   chromedriver-osx-32bit              0x000000010fe32f85 chromedriver-osx-32bit + 3329925
10  chromedriver-osx-32bit              0x000000010fe3e890 chromedriver-osx-32bit + 3377296
11  chromedriver-osx-32bit              0x000000010fe3e628 chromedriver-osx-32bit + 3376680
12  chromedriver-osx-32bit              0x000000010fe12769 chromedriver-osx-32bit + 3196777
13  chromedriver-osx-32bit              0x000000010fe3f108 chromedriver-osx-32bit + 3379464
14  chromedriver-osx-32bit              0x000000010fe26957 chromedriver-osx-32bit + 3279191
15  chromedriver-osx-32bit              0x000000010fe5ba94 chromedriver-osx-32bit + 3496596
16  chromedriver-osx-32bit              0x000000010fe7a047 chromedriver-osx-32bit + 3620935
17  libsystem_pthread.dylib             0x00007fff5b4152eb _pthread_body + 126
18  libsystem_pthread.dylib             0x00007fff5b418249 _pthread_start + 66
19  libsystem_pthread.dylib             0x00007fff5b41440d thread_start + 13


[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ValuesTest>TestRunner.beforeClass:25 » SessionNotCreated session not created: ...
[INFO] 
[ERROR] Tests run: 4, Failures: 1, Errors: 0, Skipped: 3
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 21.580 s
[INFO] Finished at: 2019-06-18T18:02:38+03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project FirstTask: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/volodymyrparlah/Documents/Learning/ssFirstTask/Task/target/surefire-reports/test-output for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
`

**Review by ghost — APPROVED** [review]

**Pull request merged**